assets/materials/shaders/BlinnPhong.hlsl

```diff
@@ -52,33 +39,33 @@ float4 psmain(VSOutput input) : SV_TARGET
         // Remove gamma
         albedo = pow(albedo, 2.2);
     }
-    
+
     float roughness = Material.roughness;
     if (Material.roughnessSelect == 1) {
         roughness = RoughnessTex.Sample(ClampedSampler, input.texCoord).r;
     }
-    
+
     float metalness = Material.metalness;
     if (Material.metalnessSelect == 1) {
         metalness = MetalnessTex.Sample(ClampedSampler, input.texCoord).r;
-    }    
-    
+    }
+
     float hardness = lerp(1.0, 100.0, 1.0 - saturate(roughness));
-    
-    float diffuse  = 0;
-    float specular = 0;
-    for (uint i = 0; i < Scene.lightCount; ++i) {    
+
+    float3 diffuse  = 0;
+    float3 specular = 0;
+    for (uint i = 0; i < Scene.lightCount; ++i) {
         float3 L = normalize(Lights[i].position - input.positionWS);
 
-        diffuse += Lambert(N, L);
-        
-        specular += BlinnPhong(N, L, V, hardness) * Lights[i].intensity;
+        diffuse += BRDF_Gouraud(N, L);
+
+        specular += BRDF_BlinnPhong(N, V, L, hardness) * Lights[i].intensity;
     }
-        
+
     float3 color = (diffuse + Scene.ambient) * albedo + specular;
-    
+
     // Reapply gamma
-    color = pow(color, 1 / 2.2); 
-    
+    color = pow(color, 1 / 2.2);
```

Comment:
Optional: should we be using ApplyGamma here? and RemoveGamma above.
Probably doesn't add much value, besides the ability to remove the comment above

Reply:
Done.